app/src/main/java/sopt/org/millie/presentation/search/bookdetail/BookDetailViewModel.kt

```diff
@@ -20,13 +22,14 @@ class BookDetailViewModel
 @Inject
 constructor(
     private val searchRepository: SearchRepository,
+    savedStateHandle: SavedStateHandle,
 ) : ViewModel() {
     private val _uiState = MutableStateFlow(BookDetailUiState())
+    private val bookId: Long = checkNotNull(savedStateHandle["bookId"])
```

Comment:
```suggestion
    val bookId = savedStateHandle.toRoute<SearchBookDetailRoute>().bookId
```
현재 checkNotNull(savedStateHandle["bookId"])로 파라미터를 가져오는 방식은 문자열 키를 직접 쓰다 보니 오타가 나도 컴파일 단계에서 잡히지 않고 결국 런타임에서 크래시로 확인하게 되는 위험이 있어용 ㅜㅜ  반면 savedStateHandle.toRoute<SearchBookDetailRoute>().bookId처럼 toRoute를 사용하면 Route의 프로퍼티를 직접 참조하기 때문에 오타가 나면 IDE에서 바로 에러를 보여주고 컴파일 자체가 되지 않아 훨씬 안전하답니다 !! 이게 바로 type-safety에요 ! 또 문자열 키를 여기저기서 쓰는 구조는 파라미터명을 바꾸거나 새로운 값을 추가할 때 모든 사용처를 직접 찾아서 수정해야 하지만 toRoute 방식은 Route 데이터 클래스만 수정하면 IDE가 사용처를 모두 리팩토링해줘서 유지보수도 훨씬 편해용 기본값 처리도 더 깔끔한데 Route에서 기본값을 정의해두면 자동으로 적용되기 때문에 매번 null 체크하거나 Elvis 연산자를 사용할 필요가 없답니다 ㅎㅎㅎ 이 패턴은 Jetpack Navigation 에서 공식적으로 권장하는 타입 세이프 네비게이션 방식이라 앞으로의 업데이트나 협업 측면에서도 더 표준적이고 안전한 구조라고 보시면 좋을 것 같아요 ~~~

그리구 _uiState랑 uiState 사이에 적지 말구 uiState 아래에 적어주면 더 가독성이 좋아질 것 같아요 !!

Reply:
자세히 설명해주셔서 감사합니다!! 덕분에 이해가 쏙쏙 되었어용